contrib/core-contract-tests/tests/pox-4/pox-4.prop.test.ts

```diff
@@ -0,0 +1,1605 @@
+            expect(actual).toBeUint(expected);
+          }
+        )
+      );
+    });
+
+    it("should return none get-reward-set-pox-address", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          fc.nat(),
+          fc.nat(),
+          (caller, index, reward_cycle) => {
+            // Arrange
+            // Act
+            const { result: actual } = simnet.callReadOnlyFn(
+              "pox-4",
+              "get-reward-set-pox-address",
+              [Cl.uint(index), Cl.uint(reward_cycle)],
+              caller
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.OptionalNone));
+            expect(actual).toBeNone();
+          }
+        )
+      );
+    });
+
+    it("should return correct get-stacking-minimum", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          (caller) => {
+            // Arrange
+            const { result: pox_4_info } = simnet.callReadOnlyFn(
+              "pox-4",
+              "get-pox-info",
+              [],
+              caller
+            );
+            assert(isClarityType(pox_4_info, ClarityType.ResponseOk));
+            assert(isClarityType(pox_4_info.value, ClarityType.Tuple));
+            const stx_liq_supply =
+              pox_4_info.value.data["total-liquid-supply-ustx"];
+            assert(isClarityType(stx_liq_supply, ClarityType.UInt));
+            const expected = Math.floor(
+              Number(stx_liq_supply.value) / TESTNET_STACKING_THRESHOLD_25
+            );
+            // Act
+            const { result: actual } = simnet.callReadOnlyFn(
+              "pox-4",
+              "get-stacking-minimum",
+              [],
+              caller
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.UInt));
+            expect(actual).toBeUint(expected);
+          }
+        )
+      );
+    });
+
+    it("should return true check-pox-addr-version for version <= 6 ", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          fc.nat({ max: 6 }),
+          (caller, version) => {
+            // Arrange
+            const expected = true;
+            // Act
+            let { result: actual } = simnet.callReadOnlyFn(
+              "pox-4",
+              "check-pox-addr-version",
+              [Cl.buffer(Uint8Array.from([version]))],
+              caller
+            );
+            // Assert
+            assert(isClarityType(actual, ClarityType.BoolTrue));
+            expect(actual).toBeBool(expected);
+          }
+        )
+      );
+    });
+
+    it("should return false check-pox-addr-version for version > 6", () => {
+      fc.assert(
+        fc.property(
+          fc.constantFrom(...simnet.getAccounts().values()),
+          fc.integer({ min: 7, max: 255 }),
```

Comment:
```suggestion
          fc.integer({ min: 7 }),
```

Reply:
Here, if I allow values greater than 255 (2^8-1), the UInt8Array will convert them to 8 bytes unsigned integer. E.g. when 266 is generated, it is converted to 0 and the function call returns `true`